source/tests/system/nirfmxwlan_driver_api_tests.cpp

```diff
@@ -339,6 +383,805 @@ TEST_F(NiRFmxWLANDriverApiTests, TXPFromExample_FetchData_DataLooksReasonable)
+
+  int32* MCSIndexArray = NULL;
+  int32 MCSIndexArrayLength = 0;
+  int32* numberOfSpaceTimeStreamsArray = NULL;
+  int32 numberOfSpaceTimeStreamsArrayLength = 0;
+  int32* spaceTimeStreamOffsetArray = NULL;
+  int32 numberOfStreamResults = INT_MIN;
+  int32 PPDUType = RFMXWLAN_VAL_OFDM_PPDU_TYPE_NON_HT;
+  int32 numberOfUsers = 0;
+  OFDMModAccFetchNumberOfUsersResponse ofdm_mod_acc_fetch_number_of_users_response;
+  OFDMModAccFetchMCSIndexResponse ofdm_mod_acc_fetch_mcs_index_response;
+  OFDMModAccFetchNumberOfSpaceTimeStreamsResponse ofdm_mod_acc_fetch_number_of_space_time_streams_response;
+  OFDMModAccFetchFrequencyErrorMeanResponse ofdm_mod_acc_fetch_frequency_error_mean_response;
+  OFDMModAccFetchSymbolClockErrorMeanResponse ofdm_mod_acc_fetch_symbol_clock_error_mean_response;
+  OFDMModAccFetchStreamRMSEVMResponse ofdm_mod_acc_fetch_stream_rmsevm_response;
+  OFDMModAccFetchStreamRMSEVMPerSubcarrierMeanTraceResponse ofdm_mod_acc_fetch_stream_rmsevm_per_subcarrier_mean_trace_response;
+  OFDMModAccFetchPilotConstellationTraceResponse ofdm_mod_acc_fetch_pilot_constellation_trace_response;
+  OFDMModAccFetchDataConstellationTraceResponse ofdm_mod_acc_fetch_data_constellation_trace_response;
+  OFDMModAccFetchCrossPowerResponse ofdm_mod_acc_fetch_cross_power_response;
+  OFDMModAccFetchIQImpairmentsResponse ofdm_mod_acc_fetch_iq_impairments_response;
+  const auto ofdm_mod_acc_fetch_composite_rmsevm_response = client::ofdm_mod_acc_fetch_composite_rmsevm(stub(), session, "", 10.0);
+  const auto ofdm_mod_acc_fetch_numberof_symbols_used_response = client::ofdm_mod_acc_fetch_numberof_symbols_used(stub(), session, "", 10.0);
+  const auto ofdm_mod_acc_fetch_ppdu_type_response = client::ofdm_mod_acc_fetch_ppdu_type(stub(), session, "", 10.0);
+  const auto ofdm_mod_acc_fetch_guard_interval_type_response = client::ofdm_mod_acc_fetch_guard_interval_type(stub(), session, "", 10.0);
+  const auto ofdm_mod_acc_fetch_lsig_parity_check_status_response = client::ofdm_mod_acc_fetch_lsig_parity_check_status(stub(), session, "", 10.0);
+  const auto ofdm_mod_acc_fetch_sigcrc_status_response = client::ofdm_mod_acc_fetch_sigcrc_status(stub(), session, "", 10.0);
+  const auto ofdm_mod_acc_fetch_sigbcrc_status_response = client::ofdm_mod_acc_fetch_sigbcrc_status(stub(), session, "", 10.0);
+  if (PPDUType == NIRFMXWLAN_INT32_OFDM_PPDU_TYPE_MU)
+  {
+    ofdm_mod_acc_fetch_number_of_users_response = client::ofdm_mod_acc_fetch_number_of_users(stub(), session, "", 10.0);
+    MCSIndexArrayLength = numberOfUsers;
+    MCSIndexArray = (int32*)malloc(MCSIndexArrayLength * sizeof(int32));
+    numberOfSpaceTimeStreamsArrayLength = numberOfUsers;
+    numberOfSpaceTimeStreamsArray = (int32*)malloc(numberOfSpaceTimeStreamsArrayLength * sizeof(int32));
+    spaceTimeStreamOffsetArray = (int32*)malloc(numberOfUsers * sizeof(int32));
+    if (MCSIndexArray == NULL || numberOfSpaceTimeStreamsArray == NULL)
```

Comment:
I'm not sure what exactly your workflow is on codegen'ing these tests. If it's important to be able to regen from the tool, that's an OK reason to leave them alone even if the style is not perfect.

That said: this if block is empty. Should probably `FAIL() << "Could not allocate array";`.

Also, I assume that the reason that we're using c-style arrays here is that it allows us to copy the code directly from the C Examples? (that's a good reason).

Reply:
Ah, yeah that was weird. I added the failure in the if block.

And yes, I believe it's more straightforward to copy over straight from the C Examples this way.

Reply:
> If it's important to be able to regen from the tool, ...

My plan is just to codegen the tests once and then throw away the generator. No need to preserve the leftover oddities.